File: factorial.f
factorial(1)

frozone() {
	x = 10
	x = factorial(x)
	print("Factorial es:")
	print(x)
}

function factorial(num) {
	on (num eq 0) do {
		return 1
	}
	next_num = num - 1
	acum = factorial(next_num)
	acum = acum * num
	return acum
}

File: factorial.c
#include <stdio.h>

int factorial(int);

int main() {
	int x = 10;
	printf("Factorial es:\n%d\n", factorial(x));
	return 0;
}

int factorial(int num) {
	if(num == 0) {
		return 1;
	}
	return num * factorial(num - 1);
}
